packages/salesforcedx-vscode-core/src/commands/templates/forceFunctionCreate.ts

```diff
@@ -47,33 +50,42 @@ export class ForceFunctionCreateExecutor extends BaseTemplateCommand {
       .build();
   }
 
-  public runPostCommandTasks(targetDir: string) {
-    if (sfdxCoreSettings.getFunctionsPullDependencies()) {
-      vscode.window.withProgress(
-        {
-          location: vscode.ProgressLocation.Window,
-          title: nls.localize(
-            'force_function_install_npm_dependencies_progress'
-          ),
-          cancellable: true
-        },
-        () => {
-          return new Promise((resolve, reject) => {
-            cp.exec('npm install', { cwd: targetDir }, err => {
-              if (err) {
-                notificationService.showWarningMessage(
-                  nls.localize(
-                    'force_function_install_npm_dependencies_error',
-                    err.message
-                  )
-                );
-                reject(err);
-              }
-              resolve();
-            });
-          });
-        }
-      );
+  public runPostCommandTasks(data: FunctionInfo) {
+    const language = data.language;
+    if (language === LANGUAGE_JAVA) {
+      const pathToSource = this.getPathToSource(data.outputdir, data.fileName);
+      const targetDir = path.join(path.dirname(pathToSource), '..', '..', '..', '..', '..');
```

Comment:
This is where I spent most time for this story. Yes, for java it is consistently 5 directories up. The alternative approach is setting java path strategy to be at this top level and we navigate 5 levels deep to open the java file at the time of creation.

Reply:
Btw, 'Should pull dependencies for java' test will be the pinning test for this behavior. If in future Functions team change this directory structure and we change the pathstrategy, this test would fail.

Reply:
can we add a comment on the test so when it breaks we know what could go wrong?